### PR TITLE
Markdown changes

### LINKED ISSUE #1277: Markdown Preview Dim Persists
@TalAmuyal Fantastic work on the markdown preview 🎉 

I've been using it a fair bit vs less elegant terminal vim plugins.
Whilst using it I noticed that after it opens it sets a `not-focused` class on the editor element which isnt removed by closing the plugin (by which I mean toggling it, not sure if that kills the process). I've found that by removing that class from the editor component the dimming goes away.

### PR DESCRIPTION
**Pull request by CrossR**
A few tiny Markdown preview changes, to make the preview experience a bit more consistent.

Fixes #1277, #2021. 

1) Close the preview when swapping out of a MD file.
2) Don't auto open the preview if it was manually closed. (Should stop it opening back up when a user closes it).
3) Fix highlighting issue (ie Preview was highlighted on Open, not Editor)
4) Fix link opening by having it use the Oni browser.

Would you be able to take a look over @TalAmuyal ?

EDIT: Now depends on a pending PR on oni-api due to the API changes for the focus / active split.

### CODE REVIEW

**Comment by CrossR** [comment]
I'm going to swap this back to WIP and add a fix for the highlighting issue as well (#1277).

**Comment by CrossR** [comment]
This PR is waiting on my other PR in `oni-api` being merged.

**Comment by bryphe** [comment]
Looks good to me, @CrossR ! Thanks for the fixes here 😄 

I just brought in the oni-api changes - bumping the version so we can get a new NPM package for it.

**Comment by bryphe** [comment]
Alright, looks like the new `oni-api` was published - `oni-api@0.0.46`.  Looks like you'll need to bump the version here. Otherwise, looks good to me! 👍

**Comment by CrossR** [comment]
Just fixing the merge conflict, then this should be fine to review/merge.

I've got a second Markdown PR to add syntax highlighting and fix relative paths for images, but I split that into a second PR as it was waiting on my PR to DefinitelyTyped to update the Marked type defs. I'll stick that up soon.

File: main/src/main.ts [hunk 1]
```diff
@@ -242,6 +242,11 @@ export function createWindow(
         Log.info("...closed event completed")
     })
 
+    currentWindow.webContents.on("will-navigate", (event, url) => {
+        event.preventDefault()
+        currentWindow.webContents.send("open-oni-browser", url)
```
Comment (bryphe) [inline main/src/main.ts]: One minor thing - we may not actually need a new `ipc` message in this case. I think we could use the `execute-command` one we have, along with the `browser.openUrl` action, something like:
currentWindow.webContents.send("execute-command", "browser.openUrl", url)

We do something similar in the menu:
https://github.com/onivim/oni/blob/16c2eb508a5c69e4cc1322bd6395a02065ca5912/main/src/menu.ts#L50

**Comment by bryphe** [comment]
Just had one minor item but not a blocker - change looks great to me! 👍 🎉  Thanks @CrossR !

**Comment by CrossR** [comment]
I'll bring this in now and address the PR comment in my other markdown changes so this can be tested a bit.

**Comment by bryphe** [comment]
Sweet! Thanks @CrossR! 💯

**Pull request merged**